entrypoints/air_pollution_shortest_path.py

```diff
@@ -57,16 +73,96 @@ def main(  # pylint: disable=too-many-arguments
     # gdf = gpd.GeoDataFrame(result_df, crs=4326, geometry="geom")
     logger.info("%s rows in hexgrid results", len(gdf))
 
-    G: nx.MultiDiGraph = ox.graph_from_address("British Library")
-    logger.info(
-        "%s nodes and %s edges in graph.", G.number_of_nodes(), G.number_of_edges()
-    )
+    if source is not None and target is not None:
+        # use bounding box of surrounding ellipse to limit graph size
+        box = ellipse_bounding_box((source[1], source[0]), (target[1], target[0]))
+        G: nx.MultiDiGraph = ox.graph_from_bbox(box[0], box[1], box[2], box[3])
 
-    logger.info("Mapping air quality predictions to the road network.")
-    G = update_cost(G, gdf, cost_attr="NO2_mean", weight_attr="length")
-    logger.debug("Printing basic stats for the graph:")
-    logger.debug(ox.stats.basic_stats(G))
+        # snap source and target to the graph
+        newSource = ox.distance.get_nearest_node(G, source)
+        newTarget = ox.distance.get_nearest_node(G, target)
 
+        logger.info(
+            "%s nodes and %s edges in graph.", G.number_of_nodes(), G.number_of_edges()
+        )
+
+        logger.info("Mapping air quality predictions to the road network.")
+        G = update_cost(G, gdf, cost_attr="NO2_mean", weight_attr="length")
+        logger.debug("Printing basic stats for the graph:")
+        logger.debug(ox.stats.basic_stats(G))
+
+        for i, (u, v, k, data) in enumerate(G.edges(keys=True, data=True)):
+            if i > 10:
+                break
+            print(u, v, k, data)
+        
+        route = nx.dijkstra_path(G, newSource, newTarget, weight="NO2_mean")
+        print(route)
 
 if __name__ == "__main__":
     typer.run(main)
+
+@app.get("/route/")
+async def get_route(source_lat:float, source_long:float,target_lat:float,target_long:float):
```

Comment:
Is pretty much a direct copy from the above main function? If so lets create a new function we re-uses the repeated functionality?